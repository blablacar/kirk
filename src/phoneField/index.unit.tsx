import React from 'react'
import renderer from 'react-test-renderer'
import { shallow, mount } from 'enzyme'
import PhoneField from 'phoneField'

const countryWhitelist = ['FR', 'ES']
const countryWhitelistUpdated = ['DE']

const customCountryNames = {
  FR: 'Custom fr label',
  ES: 'Custom es label',
}

const whitelistMapped = [
  { label: 'France +33', value: '+33' },
  { label: 'Spain (EspaÃ±a) +34', value: '+34' },
]
const whitelistMappedChange = [{ label: 'Germany (Deutschland) +49', value: '+49' }]

const defaultProps = {
  name: 'phoneField',
  onChange() {},
  countryWhitelist,
}

const customProps = {
  ...defaultProps,
  id: 'selectFieldName',
  className: 'customClass',
  ariaLabelledBy: 'labelId',
  selectFieldLabel: 'Phone Country code',
  textFieldTitle: 'Phone Number',
  textFieldPlaceholder: 'Phone number',
  defaultRegionValue: 'ES',
  defaultPhoneValue: '0765455667',
  customCountryNames,
  isInline: false,
}

describe('PhoneField', () => {
  const defaultWrapper = shallow(<PhoneField {...defaultProps} />)

  it('should have the expected markup in the DOM with default settings', () => {
    const phoneField = renderer.create(<PhoneField {...defaultProps} />).toJSON()
    expect(phoneField).toMatchSnapshot()
  })

  it('should have the expected markup in the DOM with custom settings', () => {
    const phoneField = renderer.create(<PhoneField {...customProps} />).toJSON()
    expect(phoneField).toMatchSnapshot()
  })

  it('Should have proper default state value', () => {
    expect(defaultWrapper.state()).toEqual({
      countryData: whitelistMapped,
      countryWhitelist,
      phoneNumber: '',
      phonePrefix: '',
      phoneRegionList: [],
      completePhoneNumber: '',
    })
  })

  it('Should update data when countryWhitelist changed', () => {
    defaultWrapper.setProps({ countryWhitelist: countryWhitelistUpdated })
    expect(defaultWrapper.state('countryWhitelist')).toEqual(countryWhitelistUpdated)
    expect(defaultWrapper.state('countryData')).toEqual(whitelistMappedChange)
  })

  it('should receive updated value on phone number and country code change', () => {
    const handleOnChange = jest.fn()
    const wrapper = mount(<PhoneField {...defaultProps} onChange={handleOnChange} />)
    const select = wrapper.find('select')
    const selectDOMNode = select.getDOMNode() as HTMLSelectElement
    selectDOMNode.value = '+34'
    select.simulate('change', { target: selectDOMNode })
    expect(handleOnChange).toHaveBeenCalledWith({
      name: 'phoneField',
      value: {
        completePhoneNumber: '+34',
        phonePrefix: '+34',
        phoneRegionList: ['es'],
        phoneNumber: '',
      },
    })

    const input = wrapper.find('input')
    input.simulate('change', { target: { value: '0765455667' } })
    expect(handleOnChange).toHaveBeenCalledWith({
      name: 'phoneField',
      value: {
        completePhoneNumber: '+340765455667',
        phonePrefix: '+34',
        phoneRegionList: ['es'],
        phoneNumber: '0765455667',
      },
    })
    expect(handleOnChange).toHaveBeenCalledTimes(2)
  })
})
