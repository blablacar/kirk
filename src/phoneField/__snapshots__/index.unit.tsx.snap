// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PhoneField should have the expected markup in the DOM with custom settings 1`] = `
<div
  aria-labelledby="labelId"
  className="kirk-phoneField customClass"
  id="selectFieldName"
>
  <div
    className="kirk-selectField"
  >
    <select
      aria-label="Phone Country code"
      defaultValue="+34"
      name="phonePrefix"
      onChange={[Function]}
    >
      <option
        value="+34"
      >
        Custom es label +34
      </option>
      <option
        value="+33"
      >
        Custom fr label +33
      </option>
    </select>
    <svg
      aria-hidden={true}
      className="kirk-icon"
      height={24}
      viewBox="0 0 24 24"
      width={24}
      xmlns="http://www.w3.org/2000/svg"
    >
      
      <polyline
        fill="none"
        points="9 18 15 12 9 6"
        stroke="#708C91"
        strokeLinecap="round"
        strokeLinejoin="round"
        strokeMiterlimit="10"
        strokeWidth="2"
        transform="rotate(90 12 12)"
      />
    </svg>
  </div>
  <div
    className="jsx-3265029717 kirk-textField"
  >
    <div
      className="jsx-3265029717 kirk-textField-wrapper"
    >
      <input
        className="jsx-3265029717 "
        name="phoneNumber"
        onChange={[Function]}
        placeholder="Phone number"
        title="Phone Number"
        type="tel"
        value="0765455667"
      />
      <button
        aria-hidden={true}
        className="kirk-button kirk-button-unstyled kirk-button-bubble kirk-textField-button"
        disabled={false}
        onBlur={[Function]}
        onClick={[Function]}
        onFocus={[Function]}
        onMouseDown={[Function]}
        onMouseUp={[Function]}
        onTouchEnd={[Function]}
        onTouchStart={[Function]}
        tabIndex="-1"
        type="button"
      >
        <svg
          aria-hidden={true}
          className="kirk-icon"
          height="18"
          viewBox="0 0 24 24"
          width="18"
          xmlns="http://www.w3.org/2000/svg"
        >
          
          <path
            d="M19 5L5 19M19 19L5 5"
            fill="none"
            stroke="#708C91"
            strokeLinecap="round"
            strokeLinejoin="round"
            strokeMiterlimit="10"
            strokeWidth="2"
          />
        </svg>
      </button>
    </div>
  </div>
</div>
`;

exports[`PhoneField should have the expected markup in the DOM with default settings 1`] = `
<div
  className="kirk-phoneField--inline"
>
  <div
    className="kirk-selectField"
  >
    <select
      defaultValue=""
      name="phonePrefix"
      onChange={[Function]}
    >
      <option
        value="+33"
      >
        France +33
      </option>
      <option
        value="+34"
      >
        Spain (Espa√±a) +34
      </option>
    </select>
    <svg
      aria-hidden={true}
      className="kirk-icon"
      height={24}
      viewBox="0 0 24 24"
      width={24}
      xmlns="http://www.w3.org/2000/svg"
    >
      
      <polyline
        fill="none"
        points="9 18 15 12 9 6"
        stroke="#708C91"
        strokeLinecap="round"
        strokeLinejoin="round"
        strokeMiterlimit="10"
        strokeWidth="2"
        transform="rotate(90 12 12)"
      />
    </svg>
  </div>
  <div
    className="jsx-3265029717 kirk-textField"
  >
    <div
      className="jsx-3265029717 kirk-textField-wrapper"
    >
      <input
        className="jsx-3265029717 "
        name="phoneNumber"
        onChange={[Function]}
        type="tel"
        value=""
      />
      
    </div>
  </div>
</div>
`;
